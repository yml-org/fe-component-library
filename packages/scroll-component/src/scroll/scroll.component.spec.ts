import { LitElement } from 'lit-element';

describe('scroll-component', () => {
  const SCROLL_COMPONENT = 'scroll-component';
  let scrollElement: LitElement;

  const getShadowRoot = (tagName: string): ShadowRoot =>
    document.body.getElementsByTagName(tagName)[0].shadowRoot;

    const dataSlots = [
        {
          dataStart: 1,
          dataEnd: 5,
          slotName: 'slot1',
        },
      ];

 const getSlotElements = () => {
    return getShadowRoot(SCROLL_COMPONENT)
    .querySelector('div')
    .querySelector('section')
    .querySelector('div')
 }

  beforeEach(() => {
    scrollElement = window.document.createElement(
      SCROLL_COMPONENT
    ) as LitElement;
    document.body.appendChild(scrollElement);
  });

  afterEach(() => {
    document.body.getElementsByTagName(SCROLL_COMPONENT)[0].remove();
  });

  it('matches component snapshot', async () => {
    expect(
      document.body.getElementsByTagName(SCROLL_COMPONENT)[0]
    ).toMatchSnapshot();
  });

  it('renders the slot elements', async () => {
    getShadowRoot(SCROLL_COMPONENT).querySelector('div').innerHTML =
      '<h2 class="slot-test">The scroll Component</h2>';
    await scrollElement.updateComplete;
    const elem = getShadowRoot(SCROLL_COMPONENT)
      .querySelector('div')
      .querySelector('h2');
    expect(elem.classList.contains('slot-test')).toBe(true);
  });

  it('sets the slotName property correctly', async () => {
    scrollElement['dataSlot'] = dataSlots;
    await scrollElement.updateComplete;
    const elem = getSlotElements();
    expect(elem.querySelector('slot').getAttribute('name')).toBe(dataSlots[0].slotName);
  });

  
  it('sets the dataStart property correctly', async () => {
    scrollElement['dataSlot'] = dataSlots;
    await scrollElement.updateComplete;
    const elem = getSlotElements();
    expect(elem.getAttribute('data-start')).toBe('1');
  });

  it('sets the dataStart property correctly', async () => {
    scrollElement['dataSlot'] = dataSlots;
    await scrollElement.updateComplete;
    const elem = getSlotElements();
    expect(elem.getAttribute('data-end')).toBe('5');
  });
});
