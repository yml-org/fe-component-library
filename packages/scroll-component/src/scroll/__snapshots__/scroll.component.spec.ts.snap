// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scroll-component matches component snapshot 1`] = `
P {
  "_$AL": Map {},
  "_$Do": T {
    "_$AA": <!---->,
    "_$AB": null,
    "_$AH": Et {
      "_$AD": C {
        "el": <template />,
        "parts": Array [
          Object {
            "index": 3,
            "type": 2,
          },
        ],
      },
      "_$AM": [Circular],
      "_$AN": undefined,
      "u": Array [
        T {
          "_$AA": <!--?lit$2967951902$-->,
          "_$AB": 
        ,
          "_$AH": Array [
            T {
              "_$AA": <!---->,
              "_$AB": <!---->,
              "_$AH": Et {
                "_$AD": C {
                  "el": <template />,
                  "parts": Array [
                    Object {
                      "ctor": [Function],
                      "index": 0,
                      "name": "part",
                      "strings": Array [
                        "",
                        "",
                      ],
                      "type": 1,
                    },
                    Object {
                      "ctor": [Function],
                      "index": 0,
                      "name": "data-start",
                      "strings": Array [
                        "",
                        "",
                      ],
                      "type": 1,
                    },
                    Object {
                      "ctor": [Function],
                      "index": 0,
                      "name": "data-end",
                      "strings": Array [
                        "",
                        "",
                      ],
                      "type": 1,
                    },
                    Object {
                      "ctor": [Function],
                      "index": 1,
                      "name": "name",
                      "strings": Array [
                        "",
                        "",
                      ],
                      "type": 1,
                    },
                  ],
                },
                "_$AM": [Circular],
                "_$AN": undefined,
                "u": Array [
                  M {
                    "_$AH": "slot_1",
                    "_$AM": [Circular],
                    "_$AN": undefined,
                    "element": <div
                      class="scroll-reveal text"
                      data-end="20"
                      data-start="2"
                      part="slot_1"
                    >
                      
              
                      <slot
                        name="slot_1"
                      />
                      
            
                    </div>,
                    "name": "part",
                    "options": Object {
                      "host": [Circular],
                      "isConnected": true,
                      "renderBefore": null,
                    },
                    "type": 1,
                  },
                  M {
                    "_$AH": 2,
                    "_$AM": [Circular],
                    "_$AN": undefined,
                    "element": <div
                      class="scroll-reveal text"
                      data-end="20"
                      data-start="2"
                      part="slot_1"
                    >
                      
              
                      <slot
                        name="slot_1"
                      />
                      
            
                    </div>,
                    "name": "data-start",
                    "options": Object {
                      "host": [Circular],
                      "isConnected": true,
                      "renderBefore": null,
                    },
                    "type": 1,
                  },
                  M {
                    "_$AH": 20,
                    "_$AM": [Circular],
                    "_$AN": undefined,
                    "element": <div
                      class="scroll-reveal text"
                      data-end="20"
                      data-start="2"
                      part="slot_1"
                    >
                      
              
                      <slot
                        name="slot_1"
                      />
                      
            
                    </div>,
                    "name": "data-end",
                    "options": Object {
                      "host": [Circular],
                      "isConnected": true,
                      "renderBefore": null,
                    },
                    "type": 1,
                  },
                  M {
                    "_$AH": "slot_1",
                    "_$AM": [Circular],
                    "_$AN": undefined,
                    "element": <slot
                      name="slot_1"
                    />,
                    "name": "name",
                    "options": Object {
                      "host": [Circular],
                      "isConnected": true,
                      "renderBefore": null,
                    },
                    "type": 1,
                  },
                ],
              },
              "_$AM": [Circular],
              "_$AN": undefined,
              "_$Cm": true,
              "options": Object {
                "host": [Circular],
                "isConnected": true,
                "renderBefore": null,
              },
              "type": 2,
            },
          ],
          "_$AM": [Circular],
          "_$AN": undefined,
          "_$Cm": true,
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 2,
        },
      ],
    },
    "_$AM": undefined,
    "_$AN": undefined,
    "_$Cm": true,
    "options": Object {
      "host": [Circular],
      "isConnected": true,
      "renderBefore": null,
    },
    "type": 2,
  },
  "_$E_": Promise {},
  "_$Ei": undefined,
  "_$El": null,
  "__dataSlot": Array [
    Object {
      "dataEnd": 20,
      "dataStart": 2,
      "slotName": "slot_1",
    },
  ],
  "__desktopImages": Object {
    "yonder-0001": "../assets/yonder/images/desktop/png/yonder-0001.png",
    "yonder-0002": "../assets/yonder/images/desktop/png/yonder-0002.png",
    "yonder-0003": "../assets/yonder/images/desktop/png/yonder-0003.png",
    "yonder-0004": "../assets/yonder/images/desktop/png/yonder-0004.png",
    "yonder-0005": "../assets/yonder/images/desktop/png/yonder-0005.png",
  },
  "__mobileImages": Object {
    "yonder-0001": "../assets/yonder/images/mobile/png/yonder-0001.png",
    "yonder-0002": "../assets/yonder/images/mobile/png/yonder-0002.png",
    "yonder-0003": "../assets/yonder/images/mobile/png/yonder-0003.png",
    "yonder-0004": "../assets/yonder/images/mobile/png/yonder-0004.png",
    "yonder-0005": "../assets/yonder/images/mobile/png/yonder-0005.png",
  },
  "enableUpdating": [Function],
  "hasUpdated": true,
  "isUpdatePending": false,
  "renderOptions": Object {
    "host": [Circular],
    "isConnected": true,
    "renderBefore": null,
  },
  "renderRoot": ShadowRoot {
    "_$litPart$": T {
      "_$AA": <!---->,
      "_$AB": null,
      "_$AH": Et {
        "_$AD": C {
          "el": <template />,
          "parts": Array [
            Object {
              "index": 3,
              "type": 2,
            },
          ],
        },
        "_$AM": [Circular],
        "_$AN": undefined,
        "u": Array [
          T {
            "_$AA": <!--?lit$2967951902$-->,
            "_$AB": 
        ,
            "_$AH": Array [
              T {
                "_$AA": <!---->,
                "_$AB": <!---->,
                "_$AH": Et {
                  "_$AD": C {
                    "el": <template />,
                    "parts": Array [
                      Object {
                        "ctor": [Function],
                        "index": 0,
                        "name": "part",
                        "strings": Array [
                          "",
                          "",
                        ],
                        "type": 1,
                      },
                      Object {
                        "ctor": [Function],
                        "index": 0,
                        "name": "data-start",
                        "strings": Array [
                          "",
                          "",
                        ],
                        "type": 1,
                      },
                      Object {
                        "ctor": [Function],
                        "index": 0,
                        "name": "data-end",
                        "strings": Array [
                          "",
                          "",
                        ],
                        "type": 1,
                      },
                      Object {
                        "ctor": [Function],
                        "index": 1,
                        "name": "name",
                        "strings": Array [
                          "",
                          "",
                        ],
                        "type": 1,
                      },
                    ],
                  },
                  "_$AM": [Circular],
                  "_$AN": undefined,
                  "u": Array [
                    M {
                      "_$AH": "slot_1",
                      "_$AM": [Circular],
                      "_$AN": undefined,
                      "element": <div
                        class="scroll-reveal text"
                        data-end="20"
                        data-start="2"
                        part="slot_1"
                      >
                        
              
                        <slot
                          name="slot_1"
                        />
                        
            
                      </div>,
                      "name": "part",
                      "options": Object {
                        "host": [Circular],
                        "isConnected": true,
                        "renderBefore": null,
                      },
                      "type": 1,
                    },
                    M {
                      "_$AH": 2,
                      "_$AM": [Circular],
                      "_$AN": undefined,
                      "element": <div
                        class="scroll-reveal text"
                        data-end="20"
                        data-start="2"
                        part="slot_1"
                      >
                        
              
                        <slot
                          name="slot_1"
                        />
                        
            
                      </div>,
                      "name": "data-start",
                      "options": Object {
                        "host": [Circular],
                        "isConnected": true,
                        "renderBefore": null,
                      },
                      "type": 1,
                    },
                    M {
                      "_$AH": 20,
                      "_$AM": [Circular],
                      "_$AN": undefined,
                      "element": <div
                        class="scroll-reveal text"
                        data-end="20"
                        data-start="2"
                        part="slot_1"
                      >
                        
              
                        <slot
                          name="slot_1"
                        />
                        
            
                      </div>,
                      "name": "data-end",
                      "options": Object {
                        "host": [Circular],
                        "isConnected": true,
                        "renderBefore": null,
                      },
                      "type": 1,
                    },
                    M {
                      "_$AH": "slot_1",
                      "_$AM": [Circular],
                      "_$AN": undefined,
                      "element": <slot
                        name="slot_1"
                      />,
                      "name": "name",
                      "options": Object {
                        "host": [Circular],
                        "isConnected": true,
                        "renderBefore": null,
                      },
                      "type": 1,
                    },
                  ],
                },
                "_$AM": [Circular],
                "_$AN": undefined,
                "_$Cm": true,
                "options": Object {
                  "host": [Circular],
                  "isConnected": true,
                  "renderBefore": null,
                },
                "type": 2,
              },
            ],
            "_$AM": [Circular],
            "_$AN": undefined,
            "_$Cm": true,
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 2,
          },
        ],
      },
      "_$AM": undefined,
      "_$AN": undefined,
      "_$Cm": true,
      "options": Object {
        "host": [Circular],
        "isConnected": true,
        "renderBefore": null,
      },
      "type": 2,
    },
  },
}
`;
